### PR TITLE
Vs signal gen follow up rebase6

### PR DESCRIPTION
**Pull request by dinkelbachjan**
This pull request contains all commits about follow-up work regarding the signal generator that had been added to dpsim earlier.

The models are now validated by the following two validation notebooks:
- EMT_DP_SP_Slack_PiLine_PQLoad_FrequencyRamp_CosineFM
- DP_Slack_PiLine_VSI_Ramp

In addition to that, the scalingKp parameter of the PV control has been adapted for the CIGRE MV grid in order to speed up the control.

### CODE REVIEW

File: Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 {
  "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: Please change the URLs for the reference csv files. Then, delete the code block that is marked"
```
Comment (dinkelbachjan) [inline Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb:7]: Can be removed I think

File: Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb [hunk 2]
```diff
@@ -34,12 +44,12 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    " %%bash\n",
+    "%%bash\n",
     "TOP=${TOP:-$(git rev-parse --show-toplevel)}\n",
-    "PATH=${TOP}/build/Examples/Cxx\n",
+    "PATH=${TOP}/../build/dpsim/Examples/Cxx\n",
```
Comment (dinkelbachjan) [inline Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb:49]: Needs to be fixed, looks like path for dpsim as submodule.

File: Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb [hunk 3]
```diff
@@ -157,26 +148,81 @@
...
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(12,6))\n",
+    "if emt_reference_available:\n",
+    "    ts_obj = ts_dpsim_ref_emt['v2_0']\n",
+    "    plt.plot(ts_obj.time, np.sqrt(3/2)*ts_obj.values, label='u2 (EMT, 50 µs ref)')\n",
+    "\n",
+    "ts_v2_interpolated = ts_dpsim['v2'].interpolate(50e-6)\n",
+    "ts_v2_shifted = ts.frequency_shift(ts_v2_interpolated, 50)\n",
+    "plt.plot(ts_v2_shifted.time, ts_v2_shifted.values, label='u2 (DP, 1 µs interpolated, shifted)', color='C1')\n",
```
Comment (dinkelbachjan) [inline Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb:183]: Should be 50us written here I think

File: Examples/Cxx/Circuits/DP_Slack_PiLine_VSI_with_PF_Init.cpp [hunk 4]
```diff
@@ -97,7 +97,7 @@ int main(int argc, char* argv[]) {
 	auto n2DP = SimNode<Complex>::make("n2", PhaseType::Single);
 
 	auto extnetDP = DP::Ph1::NetworkInjection::make("Slack", Logger::Level::debug);
-	extnetDP->setParameters(Complex(scenario.systemNominalVoltage,0));
+	extnetDP->setParameters(Complex(scenario.systemNominalVoltage,0), 0.0);
```
Comment (dinkelbachjan) [inline Examples/Cxx/Circuits/DP_Slack_PiLine_VSI_with_PF_Init.cpp:100]: Not required, just set for clarification I assume?

File: Examples/Cxx/Circuits/EMT_DP_SP_Slack_PiLine_PQLoad_FrequencyRamp_CosineFM.cpp [hunk 5]
```diff
@@ -80,18 +80,18 @@ void powerFlow(SystemTopology& systemPF) {
...
 	auto n1DP = SimNode<Complex>::make("n1", PhaseType::Single);
 	auto n2DP = SimNode<Complex>::make("n2", PhaseType::Single);
 
 	auto extnetDP = DP::Ph1::NetworkInjection::make("Slack", Logger::Level::debug);
-	if(waveform == "cosineFM")
+	if(waveform == "CosineFM")
 		extnetDP->setParameters(Complex(Vnom,0), 1.25, 1.25, -1.25);
 	else
-		extnetDP->setParameters(Complex(Vnom,0), 0.0, -6.25, 5.0, 0.4);
+		extnetDP->setParameters(Complex(Vnom,0), 0.0, -6.25, 0.2, 0.4, false);
```
[the remark this answers is not in the document]
Comment (rubensander) [reply Examples/Cxx/Circuits/EMT_DP_SP_Slack_PiLine_PQLoad_FrequencyRamp_CosineFM.cpp:94]: I decided to have a linear ramp in this example. Can be changed.

File: Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb (hunk 1 above)
Comment (rubensander) [reply Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb:7]: Oh yes. True dat.

File: Examples/Cxx/Examples.h [hunk 6]
```diff
@@ -279,15 +279,15 @@ namespace CIGREMV {
...
-        Real scaling_P = 10.0;
-        Real scaling_I = 1000.0;
-
-        Real KpPLL = 0.25/scaling_P;
-        Real KiPLL = 2/scaling_I;
-        Real KpPowerCtrl = 0.001/scaling_P;
-        Real KiPowerCtrl = 0.08/scaling_I;
-        Real KpCurrCtrl = 0.3/scaling_P;
-        Real KiCurrCtrl = 10/scaling_I;
+        Real scalingKp = 1;
```
Comment (dinkelbachjan) [inline Examples/Cxx/Examples.h:282]: As the change of this parameter for cigre mv is a significant change, mention it in the pull request description

File: Examples/Cxx/Circuits/DP_Slack_PiLine_VSI_with_PF_Init.cpp (hunk 4 above)
Comment (rubensander) [reply Examples/Cxx/Circuits/DP_Slack_PiLine_VSI_with_PF_Init.cpp:100]: Yes, but I'll remove it anyway.

File: Examples/Cxx/Examples.h [hunk 7]
```diff
@@ -308,10 +308,10 @@ namespace CIGREMV {
         Real phiPLLInit = 8e-06;
         Real pInit = 450000.716605;
         Real qInit = -0.577218;
-        Real phi_dInit = 3854.197405*scaling_I;
-        Real phi_qInit = -0.003737*scaling_I;
-        Real gamma_dInit = 128.892668*scaling_I;
-        Real gamma_qInit = 23.068682*scaling_I;
+        Real phi_dInit = 3854197405*scalingKi;
```
Comment (dinkelbachjan) [inline Examples/Cxx/Examples.h:311]: The new values have been determined by simulation, correct?

File: models/Include/cps/DP/DP_Ph1_VoltageSource.h [hunk 8]
```diff
@@ -54,12 +54,12 @@ namespace Ph1 {
...
 		///
 		void setSourceValue(Complex voltage);
-		///
-		//void setSourceSignal(CPS::Signal::SignalGenerator::Ptr srcSig);
-		///
-		void setParameters(Complex voltageRef, Real srcFreq = 0.0);
+		/// Setter for reference voltage without creating a signal generator
+		void setParameters(Complex voltageRef);
+		/// Setter for reference voltage and frequency with a sine wave generator
+		void setParameters(Complex voltageRef, Real srcFreq);
```
Comment (dinkelbachjan) [inline models/Include/cps/DP/DP_Ph1_VoltageSource.h:60]: Why has the default value been removed here? I think this relates to a previous comment regarding setting this value.

File: Examples/Cxx/Examples.h (hunk 7 above)
Comment (rubensander) [reply Examples/Cxx/Examples.h:311]: These new values have just been recalculated for the new definition of scalingKp and scalingKi. In fact, they have not changed.

File: models/Include/cps/DP/DP_Ph1_VoltageSource.h (hunk 8 above)
Comment (rubensander) [reply models/Include/cps/DP/DP_Ph1_VoltageSource.h:60]: Putting a default value would collide with the other function setParameters(voltageRef) which does not create a signal generator. That one could be helpful if you want to create a voltage source of which the voltage is set externally during the simulation.

File: Examples/Cxx/Circuits/EMT_DP_SP_Slack_PiLine_PQLoad_FrequencyRamp_CosineFM.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (rubensander) [reply Examples/Cxx/Circuits/EMT_DP_SP_Slack_PiLine_PQLoad_FrequencyRamp_CosineFM.cpp:94]: In addition to that, the nonlinear ramp is used in the other example so I decided to have the linear ramp here.

File: Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb [hunk 9]
```diff
@@ -1,5 +1,12 @@
 {
```
Comment (dinkelbachjan) [inline Examples/Notebooks/Circuits/DP_Slack_PiLine_VSI_Ramp.ipynb:1]: What were the reasons to keep EMT and SP out of this validation notebook?

**Comment by dinkelbachjan** [comment]
This PR is closed, since it is covered by PR #119

**Pull request closed without merging**